### PR TITLE
py-langchain: new package

### PR DESCRIPTION
**Pull request by julianolm**
# 🦜️🔗 LangChain
⚡ Building applications with LLMs through composability ⚡
Official Repo: https://github.com/langchain-ai/langchain

FYI @hyviquel

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-langchain/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    pypi = "langchain/langchain-0.0.285.tar.gz"
+
+    maintainers("julianolm")
+
+    version("0.0.285", sha256="68940dcffa63c6245e8986146614d3ce2eb22fec96ae831ed0cce97ccba3d427")
+
+    depends_on("python@3.8:", type=("build", "run"))
+
+    depends_on("py-setuptools")
+    depends_on("py-pytest")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-langchain/package.py:22]: Not required deps

File: var/spack/repos/builtin/packages/py-langchain/package.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-langchain/package.py:23]: Missing build dep:
```toml
[build-system]
requires = ["poetry-core>=1.0.0"]
```

File: var/spack/repos/builtin/packages/py-langchain/package.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    """Building applications with LLMs through composability"""
+
+    homepage = "https://www.github.com/langchain-ai/langchain"
+    pypi = "langchain/langchain-0.0.285.tar.gz"
+
+    maintainers("julianolm")
+
+    version("0.0.285", sha256="68940dcffa63c6245e8986146614d3ce2eb22fec96ae831ed0cce97ccba3d427")
+
+    depends_on("python@3.8:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-langchain/package.py:19]:
```suggestion
    depends_on("python@3.8.1:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-langchain/package.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    depends_on("py-aiohttp", type=("build", "run"))
+    depends_on("py-async-timeout", type=("build", "run"))
+    depends_on("py-pydantic", type=("build", "run"))
+    depends_on("py-dataclasses-json", type=("build", "run"))
+    depends_on("py-sqlalchemy", type=("build", "run"))
+    depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-tenacity", type=("build", "run"))
+    depends_on("py-numexpr", type=("build", "run"))
+    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-langchain/package.py:34]: Missing version requirements, see https://python-poetry.org/docs/dependency-specification/ for docs. Can you also keep these in the same order as `pyproject.toml`? Makes it easier for me to review.
```toml
[tool.poetry.dependencies]
python = ">=3.8.1,<4.0"
pydantic = ">=1,<3"
SQLAlchemy = ">=1.4,<3"
requests = "^2"
PyYAML = ">=5.3"
numpy = "^1"
dataclasses-json = "^0.5.7"
tenacity = "^8.1.0"
aiohttp = "^3.8.3"
async-timeout = {version = "^4.0.0", python = "<3.11"}
numexpr = "^2.8.4"
langsmith = "~0.0.21"
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by alecbcs** [comment]
Going to close this PR due to a lack of response after review feedback. @julianolm happy to reopen if you still want to get this merged and have time to work on it.

**Pull request closed without merging**